webboot/webboot.go

```diff
@@ -9,60 +9,146 @@ package main
 
 import (
 	"flag"
+	"fmt"
+	"io"
 	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
 
 	"github.com/u-root/webboot/pkg/dhclient"
 	"github.com/u-root/webboot/pkg/mountkexec"
 	"github.com/u-root/webboot/pkg/webboot"
 )
 
 var (
-	cmd       = flag.String("cmd", "", "Command Line")
-	mountDir  = flag.String("dir", "/tmp/mountDir", "The mount point of the ISO")
-	ifName    = flag.String("interface", "^e.*", "Name of the interface")
-	timeout   = flag.Int("timeout", 15, "Lease timeout in seconds")
-	retry     = flag.Int("retry", 5, "Max number of attempts for DHCP clients to send requests. -1 means infinity")
-	verbose   = flag.Bool("verbose", false, "Verbose output")
-	ipv4      = flag.Bool("ipv4", true, "use IPV4")
-	ipv6      = flag.Bool("ipv6", true, "use IPV6")
-	osSystems = map[string]*webboot.Distro{
-		"tinycore": &webboot.Distro{"/boot/vmlinuz64", "/boot/corepure64.gz", "console=tty1", "https:louis.com"},
-		"TestISO":  &webboot.Distro{"/kernel", "/initiso.cpi", "console=ttyS0 console=tty0", "https:louis.com"},
+	cmd      = flag.String("cmd", "", "Command Line")
+	mountDir = flag.String("dir", "/tmp/mountDir", "The mount point of the ISO")
+	ifName   = flag.String("interface", "^e.*", "Name of the interface")
+	timeout  = flag.Int("timeout", 15, "Lease timeout in seconds")
+	retry    = flag.Int("retry", 5, "Max number of attempts for DHCP clients to send requests. -1 means infinity")
+	verbose  = flag.Bool("verbose", false, "Verbose output")
+	ipv4     = flag.Bool("ipv4", true, "use IPV4")
+	ipv6     = flag.Bool("ipv6", true, "use IPV6")
+	bookmark = map[string]*webboot.Distro{
+		"tinycore": &webboot.Distro{"/boot/vmlinuz64", "/boot/corepure64.gz", "console=tty1", "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso"},
 	}
 )
 
+// parseArg takes a name and produces a filename and a URL
+// The URL can be used to download data to the file 'filename'
+// The argument is either a full URL or a bookmark.
+func parseArg(arg string) (string, string, error) {
+	var URL, filename string
+	if u, ok := bookmark[arg]; ok {
+		return u.DownloadLink, arg, nil
+	}
+
+	var err error
+	filename, err = name(arg)
+	URL = arg
+	if err != nil {
+		return "", "", fmt.Errorf("%v is not a valid URL: %v", URL, err)
+	}
+
+	return URL, filename, nil
+}
+
+// linkOpen returns an io.ReadCloser that holds the content of the URL
+func linkOpen(URL string) (io.ReadCloser, error) {
+	resp, err := http.Get(URL)
+	if err != nil {
+		return nil, err
+	}
+	// checks if request to receive URL is successful
```

Comment:
you can remove this comment, the code is very clear.

Reply:
ok, one other thing, and this is up to you.
If you look at this function carefully, you can see you no longer need the URL variable at all. 
You can remove line 51,
and on lines 53 and 56, you can just use arg instead of URL.
Further, you can remove line 44 entirely, where you declare two variables, and change line 50:
filename, err := name(arg)

I would say this is a good change to make, but you can make it on another PR if you are tired of seeing my reviews :-)

Reply:
Noo, I am not tired of seeing your reviews. Code reviews are a process. I removed the comment :)! Also, I know what you mean by the 2nd comment and what changes you are talking about. You suggested to talk about this during our 1:1 a little bit, so I will wait until then, but like we discussed, I can make that change on another PR like you suggested not because I am tired of seeing your reviews though :)